client/components/chart/index.jsx

```diff
@@ -23,196 +23,179 @@ import BarContainer from './bar-container';
-	state = {
-		data: [],
-		isEmptyChart: false,
-		maxBars: 100, // arbitrarily high number. This will be calculated by resize method
-		width: 650,
-		yMax: 0,
-		isTooltipVisible: false,
-		hasResized: false,
-	};
-
-	static propTypes = {
-		barClick: PropTypes.func,
-		data: PropTypes.array,
-		isRtl: PropTypes.bool,
-		loading: PropTypes.bool,
-		minBarWidth: PropTypes.number,
-		minTouchBarWidth: PropTypes.number,
-		numberFormat: PropTypes.func,
-		translate: PropTypes.func,
-	};
-
-	static defaultProps = {
-		barClick: noop,
-		minBarWidth: 15,
-		minTouchBarWidth: 42,
-	};
-
-	componentDidMount() {
-		if ( this.props.data && this.props.data.length && ! this.props.loading ) {
-			this.resize();
-		}
-
-		this.resize = afterLayoutFlush( this.resize );
-		window.addEventListener( 'resize', this.resize );
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( prevProps.loading && ! this.props.loading ) {
-			return this.resize();
-		}
+const isTouch = hasTouch();
 
-		if ( prevProps.data !== this.props.data ) {
-			this.updateData( this.props.data );
-		}
+/**
+ * Auxiliary method to calculate the maximum value for the Y axis, based on a dataset.
+ * @param {Array} values An array of numeric values.
+ *
+ * @return {Number} The maximum value for the Y axis.
+ */
+function getYAxisMax( values ) {
+	// Calculate max value in a dataset.
+	const max = Math.max.apply( null, values );
+	if ( 0 === max ) {
+		return 2;
 	}
 
-	componentWillUnmount() {
-		window.removeEventListener( 'resize', this.resize );
-	}
+	const log10 = Math.log10( max );
+	const sign = Math.sign( log10 );
 
-	resize = () => {
-		if ( ! this.chart ) {
-			return;
-		}
+	// Magnitude of the number by a factor fo 10 (e.g. thousands, hundreds, tens, ones, tenths, hundredths, thousandths).
+	const magnitude = Math.ceil( Math.abs( log10 ) ) * sign;
 
-		const isTouch = hasTouch();
-		const clientWidth = this.chart.clientWidth - 82;
-		const minWidth = isTouch ? this.props.minTouchBarWidth : this.props.minBarWidth;
-		const width = isTouch && clientWidth <= 0 ? 350 : clientWidth; // mobile safari bug with zero width
-		const maxBars = Math.floor( width / minWidth );
+	// Determine the base unit size, based on the magnitude of the number.
+	const unitSize =
+		sign > 0 && 1 < magnitude ? Math.pow( 10, magnitude - sign ) : Math.pow( 10, magnitude );
 
-		const updatedData = this.calculateUpdatedData( this.props.data, maxBars );
+	// Determine how many units are needed to accommodate the chart's max value.
+	const numberOfUnits = Math.ceil( max / unitSize );
 
-		this.setState( { maxBars, width, hasResized: true, ...updatedData } );
-	};
+	return unitSize * numberOfUnits;
+}
 
-	getYAxisMax = values => {
-		// Calculate max value in the dataset.
-		const max = Math.max.apply( null, values );
-		if ( 0 === max ) {
-			return 2;
+// The Chart component.
+function Chart( {
+	barClick,
+	data,
+	isRtl,
+	minBarWidth,
+	minTouchBarWidth,
+	numberFormat,
+	translate,
```

Comment:
It would be nice to put the `useTranslate` hook here. The component is already hooky... :wink:

Reply:
Oh, we also use the `numberFormat` function from the `localized` HOC 🙁 That's not ready as a hook yet.